core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -48,7 +48,7 @@ private[spark] class AppStatusListener(
 
   import config._
 
-  private var sparkVersion = SPARK_VERSION
+  private val sparkVersion = SPARK_VERSION
```

Comment:
Actually this is a bug; the version should be read from `SparkListenerLogStart` when it's in the event log. Feel free to file a separate bug.

Reply:
filed https://issues.apache.org/jira/browse/SPARK-22854 , I'll do it later